lib/src/config_provider/config_spec.dart

```diff
@@ -0,0 +1,811 @@
+      }
+    }
+
+    for (final entry in entries) {
+      final path = [...o.path, entry.key.toString()];
+      if (!inputMap.containsKey(entry.key)) {
+        // No value specified, fill in with default value instead.
+        if (entry.defaultValue != null) {
+          childExtracts[entry.key] = entry.defaultValue!
+              .call(ConfigSpecNode(path: path, value: null)) as CE;
+        } else if (entry.valueConfigSpec is FixedMapConfigSpec) {
+          final defaultValue = (entry.valueConfigSpec as FixedMapConfigSpec)
+              ._getAllDefaults(ConfigSpecNode(path: path, value: null));
+          if (defaultValue != null) {
+            childExtracts[entry.key] = (entry.valueConfigSpec
+                    as FixedMapConfigSpec)
+                ._getAllDefaults(ConfigSpecNode(path: path, value: null)) as CE;
+          }
+        }
+      } else {
+        // Extract value from node.
+        final configSpecNode =
+            ConfigSpecNode(path: path, value: inputMap[entry.key]);
+        if (!entry.valueConfigSpec._validateNode(configSpecNode, log: false)) {
+          throw ConfigSpecExtractionError(configSpecNode);
+        }
+        childExtracts[entry.key] =
+            entry.valueConfigSpec._extractNode(configSpecNode).value as CE;
+      }
+
+      if (childExtracts.containsKey(entry.key) &&
+          entry.resultOrDefault != null) {
+        // Call resultOrDefault hook for FixedMapKey.
+        entry.resultOrDefault!.call(ConfigSpecNode(
+            path: path, value: childExtracts[entry.key], nullRawValue: true));
+      }
+    }
+
+    if (additionalProperties == AdditionalProperties.Error) {
+      for (final key in inputMap.keys) {
+        if (!allKeys.contains(key)) {
+          throw ConfigSpecExtractionError(
+              o, "Invalid schema: additional properties not allowed.");
+        }
+      }
+    }
+
+    return o
+        .withValue(childExtracts, o.rawValue)
+        .transformOrThis(transform, result);
+  }
+
+  @override
+  Map<String, dynamic> _generateJsonSchemaNode(Map<String, dynamic> defs) {
+    return {
+      "type": "object",
+      if (additionalProperties != AdditionalProperties.Allow)
+        "additionalProperties": false,
+      if (schemaDescription != null) "description": schemaDescription!,
+      if (entries.isNotEmpty)
+        "properties": {
+          for (final kv in entries)
+            kv.key: kv.valueConfigSpec._getJsonRefOrSchemaNode(defs)
+        },
+      if (requiredKeys.isNotEmpty) "required": requiredKeys.toList(),
+    };
+  }
+}
+
+/// ConfigSpec for a Map that can have any number of keys.
+///
+/// [CE] typecasts result from keyValueConfigSpecs->{}->valueConfigSpec.
+///
+/// [RE] typecasts result returned by this node.
+class DynamicMapConfigSpec<CE extends Object?, RE extends Object?>
+    extends ConfigSpec<Map<dynamic, CE>, RE> {
+  /// [keyRegexp] will convert it's input to a String before matching.
```

Comment:
nit: This doesn't discribe the field, but how the keys of the map will be used.

Reply:
Updated.